File: src/main/java/com/example/Inves/models/MarketNews.java
package com.example.Inves.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.annotation.Nullable;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.net.URL;
import java.util.Date;
/**
 * @author Bossowski
 * @version 1.0
 * @email Mbossowski01@gmail.com
 * @date 28/11/2024 - 12:58
 */

@Data
@Builder
@AllArgsConstructor
public class MarketNews {


    private String Title;
    private String PublishedDate;
    private String Topic;
    private String Source;
    private String url;
}

File: Frontend/inves-v1/src/models/MarketNews.js
export class MarketNews {
    constructor(title, publishedDate, topic, source, url) {
      this.title = title;
      this.publishedDate = publishedDate;
      this.topic = topic;
      this.source = source;
      this.url = url;
    }
  
    static fromResponse(response) {
      return new MarketNews(
        response.title,
        response.publishedDate,
        response.topic,
        response.source,
        response.url
      );
    }
  }
  